addons/mail/static/src/core/common/thread_service.js

```diff
@@ -79,6 +79,7 @@ export class ThreadService {
                 thread.channelMembers.add({ ...channelMember, thread });
             }
         }
+        thread.hasFetchedMembers = true;
```

Comment:
I would move the write before the rpc, to prevent from calling it again while it is already in progress.
Note if the rpc fails, it should be reverted somehow ideally (or using a 3 step state instead of a boolean?).